### ISSUE TITLE
Coverage XML can't handle files with just newlines

### ISSUE THREAD

**Issue by rieschl** [sebastianbergmann/php-code-coverage#859]
| Q                         | A
| --------------------------| ---------------
| php-code-coverage version | 9.2.6
| PHP version               | 7.4.21
| Driver                    | Xdebug
| Xdebug version (if used)  | 3.0.4
| Installation Method       | Composer
| Usage Method              | PHPUnit
| PHPUnit version (if used) | 9.5.7

When trying to create an XML coverage report `--coverage-xml` with PHPUnit with ~some `.phtml` files~ files with just newlines in the coverage ~configuration~ directory, PHPUnit exits with error code `2` and the message:
`Generating code coverage report in PHPUnit XML format ... Call to a member function getLine() on null`.

When using another coverage report ~or removing the directory with the phtml files in the config file,~ or removing the empty file the error is gone.
I'm not quite sure how to debug this, as running PHPUnit with `--verbose --debug` doesn't print any additional lines.

This is the `coverage` section of the `phpunit.xml` file:
```
    <coverage>
        <include>
            <directory>src</directory>
            <directory suffix=".phtml">templates</directory>
        </include>
    </coverage>
```

Thanks!

Edit: edited the description to reflect the correct problem scenario

**Comment by sebastianbergmann** [comment]
Thank you for your report.

Please provide a **minimal**, **self-contained**, reproducing test case that shows the problem you are reporting.

Without such a minimal, self-contained, reproducing test case I will not be able to investigate this issue.

**Comment by rieschl** [comment]
Ah, thanks for making me try to reproduce that 🙂 
The specific scenario of the library I'm testing got me off-track. It's really much more simple than described above:
The XML coverage driver can't handle empty files (files with just a newline - completely empty files are okay), regardless of the file extension.

Consider this example:
`phpunit.xml`:
```
<phpunit>
    <testsuites>
        <testsuite name="CoverageBug">
            <directory>tests</directory>
        </testsuite>
    </testsuites>
    <coverage>
        <include>
            <directory>src</directory>
        </include>
    </coverage>
</phpunit>
```
`tests/UselessTest.php`:
```
<?php

declare(strict_types=1);

use PHPUnit\Framework\TestCase;

class UselessTest extends TestCase
{
    public function testNothing(): void
    {
        self::assertTrue(true);
    }
}
```
and a file with just a newline in `/src/`:
`src/Empty.php`
```

```
It doesn't matter if it's a `LF`,  `CRLF` or `CR` or if there are multiple newlines in the file.

That will trigger the above error:
```
$ php -dxdebug.mode=coverage vendor/bin/phpunit --coverage-xml=cov.xml
PHPUnit 9.5.7 by Sebastian Bergmann and contributors.

.                                                                   1 / 1 (100%)

Time: 00:01.011, Memory: 8.00 MB

OK (1 test, 1 assertion)

Generating code coverage report in PHPUnit XML format ... Call to a member function getLine() on null
```

Does that help or should I push a git repo with above content?
Thanks!

**Comment by sebastianbergmann** [comment]
> $ php -dxdebug.mode=coverage vendor/bin/phpunit --coverage-xml=cov.xml

Sidenote: the parameter for the `--coverage-xml` option is a directory, not a file. `cov.xml` looks to me like you are expecting it to be a file.

> Does that help or should I push a git repo with above content?

That should help. @theseer Might be easier/quicker if you have a look.

**Comment by sebastianbergmann** [comment]
I have created a reproducing test case. Thank you, @rieschl, for investing the time to minimize your test case.

Of course, the test case currently fails:

```
1) SebastianBergmann\CodeCoverage\Report\Xml\XmlTest::testForFileWithSingleEmptyLine
Error: Call to a member function getLine() on null

/usr/local/src/php-code-coverage/vendor/theseer/tokenizer/src/Tokenizer.php:122
/usr/local/src/php-code-coverage/vendor/theseer/tokenizer/src/Tokenizer.php:89
/usr/local/src/php-code-coverage/src/Report/Xml/Source.php:33
/usr/local/src/php-code-coverage/src/Report/Xml/Facade.php:174
/usr/local/src/php-code-coverage/src/Report/Xml/Facade.php:126
/usr/local/src/php-code-coverage/src/Report/Xml/Facade.php:74
/usr/local/src/php-code-coverage/tests/tests/XmlTest.php:78
/usr/local/src/php-code-coverage/vendor/phpunit/phpunit/src/Framework/TestCase.php:1341
/usr/local/src/php-code-coverage/vendor/phpunit/phpunit/src/Framework/TestCase.php:808
/usr/local/src/php-code-coverage/vendor/phpunit/phpunit/src/Framework/TestRunner.php:115
/usr/local/src/php-code-coverage/vendor/phpunit/phpunit/src/Framework/TestCase.php:547
/usr/local/src/php-code-coverage/vendor/phpunit/phpunit/src/Framework/TestSuite.php:513
/usr/local/src/php-code-coverage/vendor/phpunit/phpunit/src/Framework/TestSuite.php:513
/usr/local/src/php-code-coverage/vendor/phpunit/phpunit/src/Framework/TestSuite.php:513
/usr/local/src/php-code-coverage/vendor/phpunit/phpunit/src/TextUI/TestRunner.php:438
/usr/local/src/php-code-coverage/vendor/phpunit/phpunit/src/TextUI/Application.php:109
/usr/local/src/php-code-coverage/vendor/phpunit/phpunit/src/TextUI/Application.php:77
```

**Comment by sebastianbergmann** [comment]
This is not a bug in this library, but in one of its dependencies. I have opened https://github.com/theseer/tokenizer/issues/13 to track this.

**Comment by theseer** [comment]
> This is not a bug in this library, but in one of its dependencies. I have opened [theseer/tokenizer#13](https://github.com/theseer/tokenizer/issues/13) to track this.

Should be fixed with the just released version 1.2.1.

**Comment by rieschl** [comment]
Thanks for the quick fix!
